src/Controls/src/Core/Handlers/Items/Android/MauiRecyclerView.cs

```diff
@@ -42,6 +42,10 @@ public class MauiRecyclerView<TItemsView, TAdapter, TItemsViewSource> : Recycler
 
 		ItemTouchHelper _itemTouchHelper;
 		SimpleItemTouchHelperCallback _itemTouchHelperCallback;
+		WeakNotifyPropertyChangedProxy _layoutPropertyChangedProxy;
+		PropertyChangedEventHandler _layoutPropertyChanged;
+
+		~MauiRecyclerView() => _layoutPropertyChangedProxy?.Unsubscribe();
```

Comment:
Is it safe to trust this will always run?

Reply:
If it doesn't run, there is a leak elsewhere. It prevents the `WeakNotifyPropertyChangedProxy` from living forever.

This is the same pattern in: https://github.com/dotnet/maui/commit/0372df728628fe77fbc55b826d48279ad21d7cd5